### ISSUE TITLE
Conch mis-uses maybeDeferred in checkers.py (and maybe knownhosts.py)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10330]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#10330|
|Type|enhancement|
|Created|2022-04-11 15:25:00Z|

in `checkers.py`, `SSHPublicKeyChecker` implements `requestAvatarId` like this:

```
    def requestAvatarId(self, credentials):
        d = defer.execute(self._sanityCheckKey, credentials)
        d.addCallback(self._checkKey, credentials)
        d.addCallback(self._verifyKey, credentials)
        return d
```

and `_sanityCheckKey` like this:

```
    def _sanityCheckKey(self, credentials):
        """                                                                                                                                                                       
        Checks whether the provided credentials are a valid SSH key with a                                                                                                        
        signature (does not actually verify the signature).                                                                                                                       
                                                                                                                                                                                  
        @param credentials: the credentials offered by the user                                                                                                                   
        @type credentials: L{ISSHPrivateKey} provider                                                                                                                             
                                                                                                                                                                                  
        @raise ValidPublicKey: the credentials do not include a signature. See                                                                                                    
            L{error.ValidPublicKey} for more information.                                                                                                                         
                                                                                                                                                                                  
        @raise BadKeyError: The key included with the credentials is not                                                                                                          
            recognized as a key.                                                                                                                                                  
                                                                                                                                                                                  
        @return: the key in the credentials                                                                                                                                       
        @rtype: L{twisted.conch.ssh.keys.Key}                                                                                                                                     
        """
        if not credentials.signature:
            raise error.ValidPublicKey()

        return keys.Key.fromString(credentials.blob)
```


`_sanityCheckKey` is documented and implemented as only ever returning a result as a value, never a `Deferred` and it's private so no application code is allowed to subclass and override.

There's a similar case with `SSHPublicKeyDatabase.requestAvatarId` / `SSHPublicKeyDatabase.checkKey` - however `checkKey` isn't private so ... maybe it's stuck as-is.

And, likewise, there's `KnownHostsFile.verifyHostKey` / `KnownHostsFile.hasHostKey`.  `hasHostKey`, again, is public though ... 


<details><summary>Searchable metadata</summary>

```
trac-id__10330 10330
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1649690700163524 1649690700163524
changetime__1649702915866006 1649702915866006
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

https://github.com/twisted/twisted/pull/1716

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `closed`|
|-|-|

Merged in fac92c38c8b5cc5c4a99d5c71d8f12fc7c3b9e28
